Mathlib/RingTheory/PowerSeries/WellKnown.lean

```diff
@@ -64,6 +65,87 @@ theorem map_invUnitsSub (f : R →+* S) (u : Rˣ) :
 
 #align power_series.map_inv_units_sub PowerSeries.map_invUnitsSub
 
+section invOneSubPow
+
+variable {S : Type _} [CommRing S] (d : ℕ)
+
+theorem mk_one_pow_eq_mk_choose_add :
+    (@mk S 1) ^ (d + 1) = @mk S fun n => Nat.choose (d + n) d := by
+  induction d with
+  | zero => simp only [Nat.zero_eq, zero_add, pow_one, Nat.choose_zero_right, Nat.cast_one]; rfl
```

Comment:
`simp` shows you what it can't get past by showing you its output. Here the problem is that it doesn't know that a good way to check that two functions are equal is to check they're equal on all inputs (because sometimes that's not the right direction so it's not a simp lemma). Here you want the `ext` tactic, and then `simp` can do the rest.
```suggestion
  | zero => ext; simp
```